library/src-3.x/scala/TupledFunction.scala

```diff
@@ -0,0 +1,16 @@
+package scala
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("${F} cannot be tupled as ${Args} => ${R}")
+trait TupledFunction[F, Args <: Tuple, R] {
```

Comment:
Maybe add more documentation about the relationship between the `F`, `Args` and `R` types?

Reply:
Added documentation